diff --git a/config/version.mk b/config/version.mk
index e2d8d41..37dda28 100644
--- a/config/version.mk
+++ b/config/version.mk
@@ -22,7 +22,7 @@ TARGET_PRODUCT_SHORT := $(subst omni_,,$(TARGET_PRODUCT_SHORT))
 
 # Build the final version string
 ifdef BUILDTYPE_RELEASE
-    ROM_VERSION := $(PLATFORM_VERSION)-$(TARGET_PRODUCT_SHORT)
+    ROM_VERSION := $(PLATFORM_VERSION)_$(HT_BUILD_NUMBER)-$(TARGET_PRODUCT_SHORT)
 else
 ifeq ($(ROM_BUILDTIME_LOCAL),y)
     ROM_VERSION := $(PLATFORM_VERSION)-$(shell date +%Y%m%d-%H%M%z)-$(TARGET_PRODUCT_SHORT)-$(ROM_BUILDTYPE)
@@ -33,5 +33,5 @@ endif
 
 # Apply it to build.prop
 PRODUCT_PROPERTY_OVERRIDES += \
-    ro.modversion=OmniROM-$(ROM_VERSION) \
+    ro.modversion=OmniToad-$(ROM_VERSION) \
     ro.omni.version=$(ROM_VERSION)
